sklearn/utils/tests/test_array_api.py

```diff
@@ -90,6 +91,18 @@ def test_get_namespace_array_api():
         with pytest.raises(TypeError):
             xp_out, is_array_api_compliant = get_namespace(X_xp, X_np)
 
+        def mock_getenv(key):  # pragma: nocover
```

Comment:
This should be covered right, since it is used in the monkeypatch below?

I would rather have codecov red and be like "oh well codecov, sometimes it is hard to understand you, whatever :shrug:" than sprinkle a few comments like this that don't seem to make too much sense even if it makes codecov happy ...

Reply:
The if condition was making the patch codecoverage CI fail. We can remove it, if it's okay to ignore it.

Reply:
I think this is OK to ignore it.

For more details, I am guessing what coverage is telling you is that the `if` line is partially covered because we are using branch coverage which is more thorough than statement coverage (`branch=True` in `.coveragerc`), i.e. the condition in the if line is never False.

Reply:
Okay let me revert.